### ISSUE TITLE
Reloading requires restarting rollup and extension

### ISSUE THREAD

**Issue by stevenheidel** [extend-chrome/rollup-plugin-chrome-extension#34]
**Describe the bug**

In order to get my code to reload in either the popup window or the background page, I have to both:
- Ctrl+C and restart rollup
- flip the extension from disabled to enabled

If I don't do either of those steps then my new code is not actually run. 

**To Reproduce**

The following rollup.config.js:
```js
export default {
    input: 'src/manifest.json',
    output: {
        dir: 'dist',
        format: 'esm',
        sourcemap: true,
    },
    plugins: [
        chromeExtension(),
        simpleReloader(),
        typescript({include: 'src/**/*.ts', sourceMap: true}),
        resolve(),
        commonjs(),
    ],
}
```

The following manifest.json (revelant sections):
```json
  "background": {
    "scripts": [
      "background/background.ts"
    ],
    "persistent": false
  },
  "browser_action": {
    "default_icon": "assets/icon.png",
    "default_popup": "popup/popup.html"
  },
```
(Note: popup.html has a script tag pointing to popup.ts)

Run `rollup -c -w`

**Expected behavior**

For popup: I expect that after changing the code, it will rebuild and then next time I click the popup the new code will run.

For background page: I expect that after changing the code, it will restart the background page with the new code.

**Versions:**
 - OS: MacOS Catalina
 - Browser: Chrome 83

    "@rollup/plugin-commonjs": "^11.1.0",
    "@rollup/plugin-node-resolve": "^8.0.1",
    "@rollup/plugin-typescript": "^4.1.2",
    "rollup": "^1.32.1",
    "rollup-plugin-chrome-extension": "^2.2.1",
    "typescript": "^3.7.4",
    "webextension-polyfill-ts": "^0.12.0"

**Additional context**
Love the minimal setup of this build, hoping we can get it to work 🤞

**Comment by jacksteamdev** [comment]
@stevenheidel Thanks for the great issue! 

I ran into this last week... it's a bug in [`@rollup/plugin-typescript@4.1.2`](`@rollup/plugin-typescript@4.1.2`), which [should be fixed soon](https://github.com/rollup/plugins/pull/425). 

Can you try downgrading to `@rollup/plugin-typescript@^3` and let me know if that works in the meantime?

**Comment by stevenheidel** [comment]
Yep that works! 🎉 

Now I don't have to restart the `rollup` command anymore. Also the popup gets the new code after a code change.

Only thing that's missing now is that the background page isn't reloading until I click the refresh button on chrome://extensions (unsure what the expected behavior is supposed to be for the background page)

**Comment by jacksteamdev** [comment]
@stevenheidel Thanks for the tea!

The background page should reload itself, I'll look into it.

**Comment by jacksteamdev** [comment]
@stevenheidel Can you provide a minimal repo that reproduces what you're seeing? I haven't been able to reproduce this problem using `@rollup/plugin-typescript@3.1.1` and `rollup-plugin-chrome-extension@2.2.1`.

I _did_ run into a problem where an old lock file was interfering with installing the correct versions. `npm list` will show what versions are actually installed (not always what is in `package.json`).

**Comment by jacksteamdev** [comment]
For those who may be interested...

### How the simple reloader works

In a Chrome extension, the popup and background pages are like iframes in the extension process. They're called _privileged pages_. The simple reloader makes the background page fetch a timestamp file on an interval. We write a new timestamp file after Rollup finishes writing its files.

When the timestamp changes, we use `chrome.runtime.reload()` to reload the extension process with the new code. It's the same as clicking the reload button in `chrome://extensions`.

The simple reloader only adds code to the background page and the content scripts. The old popup just closes when the extension reloads. The background page gets the new code immediately. When you open the popup or options page again, it gets the new code.

**Comment by stevenheidel** [comment]
Here's a minimal repo: https://github.com/stevenheidel/rollup-plugin-chrome-extension-issue

Steps to reproduce:
1. Run `npm run start`
2. Load unpacked dist/ folder into chrome
3. Make a change to background/background.ts
4. Observe in the background console that no reload happens

If I press the refresh button on the chrome extensions page, then both the background and content-script will refresh.

**Comment by stevenheidel** [comment]
Potentially relevant, I also get this error in the background console:
`bg-reloader-client-488ca625.js:1 Failed to load resource: net::ERR_FILE_NOT_FOUND`

**Comment by jacksteamdev** [comment]
@stevenheidel This should be fixed in v3.0.1. Please let me know if it still doesn't work for you. Thanks for your support, patience and help!

**Comment by stevenheidel** [comment]
Just upgraded - it's working great! 🙏
